src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```diff
@@ -120,6 +121,39 @@ public void InitializeService(ServiceHost serviceHost)
             this.SessionMonitor.AddSessionListener(this);
         }
 
+        /// <summary>
+        /// Handle request to start a profiling session
+        /// </summary>
+        internal async Task HandleStartProfilingRequest(StartProfilingParams parameters, RequestContext<StartProfilingResult> requestContext)
```

Comment:
While you're refactoring this anyways, I would strongly suggest taking the time to add in debug logging for key events (start stream, stop stream, errors, received stream data, etc.) Basically anything that would be helpful in debugging an issue a customer has when you don't have access to their machine.

I'd probably suggest doing this as a separate follow up PR just to make it easier to review and not add more churn to this one, but getting started by adding comments in your local environment would be a good idea.

Reply:
@smartguest Just to check, are you planning on doing this? It'd be hugely valuable and this is something I feel we should get in before pushing this to stable so that if customers start having issues we can quickly and easily track down where issues may be occurring. 

@kburtram FYI

Reply:
I plan on getting this into insiders by December at the very least.

Reply:
So keep in mind that if you check this in and don't want it in stable then you're essentially blocking anyone else from making changes in STS that they want to get into stable. So if you don't want to ship this in stable then we'd need to discuss when this should actually be merged in.